File: src/main/java/spring/webmvc/infrastructure/persistence/adapter/MemberRepositoryAdapter.java
package spring.webmvc.infrastructure.persistence.adapter;

import java.util.Optional;

import org.springframework.stereotype.Component;

import lombok.RequiredArgsConstructor;
import spring.webmvc.domain.model.entity.Member;
import spring.webmvc.domain.repository.MemberRepository;
import spring.webmvc.infrastructure.persistence.MemberJpaRepository;

@Component
@RequiredArgsConstructor
public class MemberRepositoryAdapter implements MemberRepository {

	private final MemberJpaRepository jpaRepository;

	@Override
	public Optional<Member> findById(Long id) {
		return jpaRepository.findById(id);
	}

	@Override
	public Optional<Member> findByAccount(String account) {
		return jpaRepository.findByAccount(account);
	}

	@Override
	public boolean existsByAccount(String account) {
		return jpaRepository.existsByAccount(account);
	}

	@Override
	public Member save(Member member) {
		return jpaRepository.save(member);
	}

	@Override
	public void delete(Member member) {
		jpaRepository.delete(member);
	}
}

File: spring-web-kotlin/src/main/kotlin/spring/web/kotlin/infrastructure/persistence/adapter/MemberRepositoryAdapter.kt
package spring.web.kotlin.infrastructure.persistence.adapter

import org.springframework.data.repository.findByIdOrNull
import org.springframework.stereotype.Repository
import spring.web.kotlin.domain.model.entity.Member
import spring.web.kotlin.domain.repository.MemberRepository
import spring.web.kotlin.infrastructure.persistence.MemberJpaRepository

@Repository
class MemberRepositoryAdapter(
    private val jpaRepository: MemberJpaRepository
) : MemberRepository {
    override fun findByIdOrNull(id: Long) = jpaRepository.findByIdOrNull(id)

    override fun findByAccount(account: String) = jpaRepository.findByAccount(account)

    override fun existsByAccount(account: String) = jpaRepository.existsByAccount(account)

    override fun save(member: Member) = jpaRepository.save(member)

    override fun delete(member: Member) {
        jpaRepository.delete(member)
    }
}